prow/gerrit/adapter/adapter.go

```diff
@@ -166,12 +174,39 @@ func (c *Controller) applyGlobalConfigOnce(cfg config.Getter, gerritClient *clie
 	gerritClient.Authenticate(cookiefilePath, tokenPathOverride)
 }
 
+// Helper function to create the cache used for InRepoConfig. Currently only attempts to create cache and returns nil if failed.
+func createCache(cloneURI *url.URL, cookieFilePath string, cacheSize int, configAgent *config.Agent) (cache *config.InRepoConfigCache, err error) {
+	opts := git.ClientFactoryOpts{
+		CloneURI:       cloneURI.String(),
+		Host:           cloneURI.Host,
+		CookieFilePath: cookieFilePath,
+	}
+	gc, err := git.NewClientFactory(opts.Apply)
+	if err != nil {
+		//TODO(mpherman): Return Error once we know this WAI
+		logrus.Warn("Failed to create Gerrit Client for InRepoConfig")
+		return nil, nil
+	}
+	// Initialize cache for fetching Presubmit and Postsubmit information. If
+	// the cache cannot be initialized, exit with an error.
+	cache, err = config.NewInRepoConfigCache(
+		cacheSize,
+		configAgent,
+		config.NewInRepoConfigGitCache(gc))
+	// If we cannot initialize the cache, exit with an error.
+	if err != nil {
```

Comment:
Looking at the callstack of `config.NewInRepoConfigCache`, the only case where err is non nil is when `cacheSize < 0`, I think we should just error out here

Reply:
Done